paymentsheet/src/test/java/com/stripe/android/paymentsheet/ui/DefaultEditPaymentMethodViewInteractorTest.kt

```diff
@@ -279,12 +282,35 @@ class DefaultEditPaymentMethodViewInteractorTest {
         }
     }
 
+    @Test
+    fun `availableBrands should be filtered according to cardBrandFilter`() = runTest {
+        val cardBrandFilter = object : CardBrandFilter {
+            override fun isAccepted(cardBrand: CardBrand): Boolean {
+                return cardBrand == CardBrand.Visa
+            }
+
+            override fun describeContents(): Int = 0
+
+            override fun writeToParcel(parcel: Parcel, flags: Int) {
+                // no-op
+            }
+        }
+
+        val interactor = createInteractor(cardBrandFilter = cardBrandFilter)
+
+        interactor.viewState.test {
+            val state = awaitItem()
+            assertThat(state.availableBrands).isEqualTo(listOf(VISA_BRAND_CHOICE))
```

Comment:
Can you add another test that ensures we don't filter this out if isAccepted always returns true?

Reply:
Added!